ubuntu_cloud_image_changelog/ubuntu_cloud_image_changelog/lib.py

```diff
@@ -148,66 +149,81 @@ def parse_changelog(
+
+    try:
+        parseable_changelog_diff = get_parseable_changelog_diff(from_changelog_filename, to_changelog_filename)
+        parsed_changelog = Changelog(parseable_changelog_diff)
+        # The changelog blocks are in reverse order; we'll see high|to before low|from.
+        for changelog_block in parsed_changelog:
+            if not changelog_block.changes():
+                continue
+            if changelog_block.version and \
+                Version(changelog_block.version.full_version) > \
+                Version(to_version):
+                # if changelog_block.version is None then this is a
+                # changelog we are unable to parse
+                logging.warning("Changelog block version {} is unexpectedly greater than to_version {}".format(
+                    changelog_block.version.full_version, to_version))
+
+            # Attempt to parse theCVEs referenced in the changelog entries
+            cves = []
+            if highlight_cves:
+                cves = _parse_cve_details(changelog_block.changes(), launchpad)
+
+            changelog_change = Change(
+                package=changelog_block.package,
+                version=str(changelog_block.version),
+                urgency=changelog_block.urgency,
+                distributions=changelog_block.distributions,
+                launchpad_bugs_fixed=changelog_block.lp_bugs_closed,
+                author=changelog_block.author,
+                date=changelog_block.date,
+                log=changelog_block.changes(),
+                cves=cves,
+            )
+            changelogs.append(changelog_change)
+            if count and len(changelogs) == count:
+                break  # we have enough blocks now
+
+        # log a warning if we have no changelog
+        if not changelogs:
+            logging.warning(
+                "Unable to parse changelog diff from files {} to {}".format(
+                    from_changelog_filename, to_changelog_filename
                 )
-        except Exception as ex:
-            raise ex
-            raise Exception("Unable to parse package changelog {}".format(changelog_filename))
-
-        return changelogs
+            )
+    except Exception as ex:
+        logging.exception(ex)
+        raise ex
+
+    return changelogs
+
+
+def get_parseable_changelog_diff(
+    from_changelog_filename: Optional[str],
+    to_changelog_filename: str,
+) -> str:
+    if not from_changelog_filename:
+        return open(to_changelog_filename).read()
+    try:
+        with open(from_changelog_filename, "r") as from_changelog_fileptr, \
+        open(to_changelog_filename, "r") as to_changelog_fileptr:
+            changelog_diff_lines = []
+            unified_diff = difflib.unified_diff(from_changelog_fileptr.readlines(), to_changelog_fileptr.readlines())
+            for line in unified_diff:
+                if line.startswith(("+++", "---", "@@")):
```

Comment:
Can you add comments as to why we are filtering on these characters?

Reply:
Done

Reply:
Thank you for the additional comments